src/utils/ReadOnlyException.ts

```diff
@@ -0,0 +1,10 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License.
+
+import { InternalException } from './InternalException';
```

Comment:
Let's try to use (or at least mimic) native javascript readonly fields. For example, I did some digging and you can make an existing property on an object readonly like this:
```
Object.defineProperty(appStartContext, 'appHookData', { writable: false });
```

The error is of type `TypeError` and the message is something like this:

> Cannot assign to read only property 'hookData'...

Reply:
Were you able to reproduce this error message? In my testing, I saw that setting a property to readonly will only cause it to ignore set statements, but I didn't see an error being thrown. Anyway, I can use that message in my error

Reply:
Yes I was able to reproduce. I believe this error is only thrown if you're running in `strict` mode, but I thought TypeScript output `"use strict";` in all our js files by default

Anyways I don't have a strong preference as long as we're exactly replicating node.js behavior

Reply:
Ah that makes sense. I was only testing on the node CLI